docs/walkthrough.md

```diff
@@ -0,0 +1,320 @@
+...
+```
+9. Now that your environment is "locked," you can test this by creating a new environment from the `conda-lock.yml` file.
+```bash
+(env-management-demo) $ conda deactivate
+(base) $
+(base) $ conda-lock install -n env-locked
+INFO:root:Downloading and Extracting Packages
+INFO:root:numpy-1.24.1
+INFO:root:pandas-1.5.3
+...
+(base) $ conda activate env-locked
+(env-locked) $
+```
+10. After testing this environment on your code, you can share the `conda-lock.yml` file with other developers and have them run the code with the same `conda-lock` command.
+
+# Workflow 2: Locking an Existing Project (TBD)
+While this part of the tutorial is unfinished, you could get a pretty good idea of how to do this without this section. For now, you can continue to [Reproduce](#reproduce). Good luck!
+
+# Reproduce
+Once you have the environment locked into a `conda-lock.yml` multi-platform file, you will be able to reproduce the environment (i.e., create and activate it) with the following command as long as the file is in your current directory:
+```bash
+(base) $ conda-lock install -n NAME
+(base) $ conda activate NAME
+(NAME) $
+```
+## With Docker in Mind
+To minimize the size of the lockfile, you may want to render a single-platform lockfile for linux.
+```bash
+(base) $ conda-lock render -p linux-64
+...
+```
+The default file result is a file named `conda-linux-64.lock`. This lockfile can then be copied into an image and used from within the image.
+```
+conda create -n my-locked-env --file conda-linux-64.lock
+```
+
+# Upgrade
+Most work related to upgrading your environment requires manually updating your `environment.yml` you created in an earlier step. In the following examples, I will use the existing `environment.yml` to demonstrate upgrading.
+
+## Upgrade an existing package
+Say there is a new version of pandas available and you want to update to the latest one. This is the simplest use case.
+```bash
+(base) $ conda-lock --update pandas
+```
+
+## Adding a New Package via `conda`
+We have `pandas` installed, but of course we want to use `matplotlib`.
+1. Install `matplotlib`
+```bash
+(base) $ conda install -c conda-forge matplotlib
+...
+```
+2. Add `matplotlib` to the `environment.yml` file.
+```yml
+# environment.yml
+channels:
+  - conda-forge
+  - defaults
+dependencies:
+  - python
+  - pandas
+  - matplotlib # NEW
+platforms:
+  - linux-64
+  - osx-64
+  - win-64
+  - osx-arm64
+```
+3. Update the file using `conda-lock`
+```bash
+(base) $ conda-lock --update matplotlib
+Locking dependencies for ['linux-64', 'osx-64', 'osx-arm64', 'win-64']...
+...
+```
+Note: If you update to a specific version of a package, you will need to provide the version in the command. For example, `matplotlib=3.5` should be included after the `--update` flag. This update flag is useful for the following:
+* Updating to the latest version of a package (no version specified)
+* Updating a package to a specific version
+* Adding a new package
```

Comment:
Does this actually work?  I am trying to do this same type of thing.  In my case, I want to add `fsspec` to https://github.com/firesim/firesim/blob/main/conda-reqs.conda-lock.yml.  To do that, I:
1. add `fsspec` to https://github.com/firesim/firesim/blob/main/conda-reqs.yaml
2. `conda-lock -f conda-reqs.yaml -p linux-64 --lockfile conda-reqs.conda-lock.yml --update fsspec`

NOTE: I don't necessarily install the package into the conda environment because conda-lock doesn't use the contents of the currently active environment  in  it's calculation of the lockfile, does it?

However, I find that unless package you're trying to add is in the lockfile, the `intersection()` at https://github.com/conda/conda-lock/blob/854fca9923faae95dc2ddd1633d26fd6b8c2a82d/conda_lock/conda_solver.py#L151 filters the added package out of `to_update` and `conda-lock` effectively re-solves the entire environment, it doesn't use the `update` code.

Also, I think it is mildly confusing to suggest step 1 is installing the package into the currently active environment because it implies that having the packages installed somehow influences how conda-lock creates the lockfile, which I don't think is correct.  If you are using a lockfile to reproducibly create an environment, it would probably be better to create the updated lockfile and then use it to create the environment.  I haven't tried installing on top of an existing environment to see what happens but I'd expect it to be the same as removing the environment completely and reinstalling everything in the lockfile (but perhaps with optimizations for leaving unchanged packages alone).

Reply:
The `INFO` logger line after `Locking dependencies for ...` is indicative of whether update is happening or not.

When update is being used, it should say something like:
```
INFO:conda_lock.conda_solver:to_update: ['matplotlib']
```

I suspect the output looks like :
```
INFO:conda_lock.conda_solver:linux-64 using specs ['python', 'pandas', 'matplotlib']
```
and that indicates that the entire environment is being resolved and `--update` is effectively being ignored.

Reply:
@mariusvniekerk, reading https://github.com/conda/conda-lock/issues/158 it seems like incremental updating should be possible.  If you could take a look at this use-case and comment on what I've found, it would be really helpful.

Reply:
> 

I can confirm that what is shown in the walkthrough is in fact `..using specs...` and appears to be ignoring the `--update` flag. I can be sure to remove this section, or at least clarify that the environment is being solved from scratch.

Reply:
I'm not very familiar with the `--update` flag, but if there's a suspected bug (or a missing warning message), let's open an issue for follow-up.

Reply:
@timsnyder-siv do you want to investigate this further and open an issue for it? It looks like #158 does not mention the `--update` flag. I see mentions of `mamba` and "performance" but I think what we are discussing might be different?

Reply:
In general the update flag should only really be used if you absolutely have to minimize the change to a lock.  The more standard approach is to just delete and get a new solve

Reply:
Thanks @mariusvniekerk.  I guess I was asking whether it is supposed to be possible to add new specs to the solve and ask to only update those new specs or if one currently must re-solve the whole environment.

Reply:
That should work i think, but yeah, probably worth testing if it behaves in the way that you'd expect

Reply:
> The more standard approach is to just delete and get a new solve

I can modify the document to recommend this approach. However, to your point,

> That should work i think, but yeah, probably worth testing if it behaves in the way that you'd expect

It appears the `--update` flag is not behaving as expected. For example, when I used it to add `matplotlib` it behaved like this:

```bash
(base) $ conda-lock --update matplotlib
INFO:conda_lock.conda_solver:linux-64 using specs ['python', 'pandas', 'matplotlib']
``` 

This is in contrast to what was expected according to @timsnyder-siv:

```bash
INFO:conda_lock.conda_solver:to_update: ['matplotlib']
```

So it appears the command is ignoring the flag.

Reply:
@jesshart we should probably open a new issue referencing this.  If we're feeling strong in our python-fu, PR'ing a test would also be helpful and if we're feeling particularly masterful, possibly a patch that helps conda-lock do this solve.  I looked at the code enough to think I might be able to come up with a patch but it will probably take me longer to get right than I have time for right now.  I'll open a ticket from this thread.